src/core/border_router/routing_manager.cpp

```diff
@@ -0,0 +1,700 @@
+ *
+ */
+
+#include "border_router/routing_manager.hpp"
+
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+
+#include <string.h>
+
+#include <openthread/platform/infra_if.h>
+
+#include "common/code_utils.hpp"
+#include "common/debug.hpp"
+#include "common/instance.hpp"
+#include "common/locator-getters.hpp"
+#include "common/logging.hpp"
+#include "common/random.hpp"
+#include "common/settings.hpp"
+#include "net/ip6.hpp"
+#include "thread/network_data.hpp"
+#include "thread/network_data_leader.hpp"
+#include "thread/network_data_local.hpp"
+#include "thread/network_data_notifier.hpp"
+
+namespace ot {
+
+RoutingManager::RoutingManager(Instance &aInstance)
+    : InstanceLocator(aInstance)
+    , mInfraIfIndex(0)
+    , mRouterAdvertisementTimer(aInstance, HandleRouterAdvertisementTimer, this)
+    , mRouterSolicitTimer(aInstance, HandleRouterSolicitTimer, this)
+    , mDiscoveredOnLinkPrefixInvalidTimer(aInstance, HandleDiscoveredOnLinkPrefixInvalidTimer, this)
+{
+    memset(mInfraIfName, 0, sizeof(mInfraIfName));
+
+    mLocalOmrPrefix.Clear();
+    mAdvertisedOmrPrefix.Clear();
+
+    mLocalOnLinkPrefix.Clear();
+    mAdvertisedOnLinkPrefix.Clear();
+    mDiscoveredOnLinkPrefix.Clear();
+}
+
+otError RoutingManager::Init(uint32_t aInfraIfIndex, const char *aInfraIfName)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(!IsInitialized(), error = OT_ERROR_ALREADY);
+    VerifyOrExit(aInfraIfIndex > 0 && strlen(aInfraIfName) < kMaxInfraIfNameLength, error = OT_ERROR_INVALID_ARGS);
+
+    strcpy(mInfraIfName, aInfraIfName);
+    mInfraIfIndex = aInfraIfIndex;
+
+    LoadOrGenerateRandomOmrPrefix();
+    LoadOrGenerateRandomOnLinkPrefix();
+
+exit:
+    return error;
+}
+
+void RoutingManager::LoadOrGenerateRandomOmrPrefix(void)
+{
+    if (Get<Settings>().ReadOmrPrefix(mLocalOmrPrefix) != OT_ERROR_NONE || !IsValidOmrPrefix(mLocalOmrPrefix))
+    {
+        Ip6::NetworkPrefix randomOmrPrefix;
+
+        otLogNoteBr("no valid OMR prefix found in settings, generating new one");
+
+        if (randomOmrPrefix.GenerateRandomUla() != OT_ERROR_NONE)
+        {
+            otLogCritBr("failed to generate random OMR prefix");
+            ExitNow();
+        }
+
+        mLocalOmrPrefix.Set(randomOmrPrefix);
+        IgnoreError(Get<Settings>().SaveOmrPrefix(mLocalOmrPrefix));
+    }
+
+exit:
+    return;
+}
+
+void RoutingManager::LoadOrGenerateRandomOnLinkPrefix(void)
+{
+    if (Get<Settings>().ReadOnLinkPrefix(mLocalOnLinkPrefix) != OT_ERROR_NONE ||
+        !IsValidOnLinkPrefix(mLocalOnLinkPrefix))
+    {
+        Ip6::NetworkPrefix randomOnLinkPrefix;
+
+        otLogNoteBr("no valid on-link prefix found in settings, generating new one");
+
+        if (randomOnLinkPrefix.GenerateRandomUla() != OT_ERROR_NONE)
+        {
+            otLogCritBr("failed to generate random on-link prefix");
+            ExitNow();
+        }
+
+        randomOnLinkPrefix.m8[6] = 0;
+        randomOnLinkPrefix.m8[7] = 0;
```

Comment:
It is a requirement in CHIP addressing proposal. I an okay to remove this for now since the proposal is not final. Do you have a strong preference on this?

Reply:
SGTM